### PR TITLE
[MNT] Conda Forge 0.13.1 Release 

### PR DESCRIPTION
**Pull request by TNTran92**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Things I have done in this PR:
Set build version to 0.13.1
Set SHA to SHA256 found on [PyPi](https://pypi.org/project/sktime/0.13.1/#modal-close)
Set build number to 0

Add the following soft dependencies
- gluonts >=0.9.0
- kotsu >=0.3.1
- tensorflow
- tensorflow-probability
- tslearn >=0.5.2
- statsforecast >=0.5.2
- dtw-python
- esig ==0.9.7
- mne
- pyod >=0.8.0

### CODE REVIEW

**Comment by TNTran92** [comment]
@conda-forge-admin

**Review by fkiraly — CHANGES_REQUESTED** [review]
I think the dependencies are not quite correct for 0.13.1?
E.g., we did not add `backoff`.

Also, the builds are failing, not sure why though.

**Comment by TNTran92** [comment]
@conda-forge-admin, please rerender

**Comment by lmmentel** [comment]
It seems the link to the github release assets changed. The assets are now under https://github.com/alan-turing-institute/sktime/archive/refs/tags/v0.13.1.tar.gz url so the template in the `meta.yml` would need to be updated.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ package:
 
 source:
   url: https://github.com/alan-turing-institute/{{ name }}/archive/v.{{ version }}.tar.gz
```
Comment (dhirschfeld) [inline recipe/meta.yaml:10]:
```suggestion
  url: https://github.com/alan-turing-institute/{{ name }}/archive/v{{ version }}.tar.gz
```

**Comment by fkiraly** [comment]
> It seems the link to the github release assets changed. The assets are now under https://github.com/alan-turing-institute/sktime/archive/refs/tags/v0.13.1.tar.gz url so the template in the `meta.yml` would need to be updated.

Ah, could that be the reason why the automated releases on conda are all failing?

**Comment by dhirschfeld** [comment]
The correct url has now revealed a SHA256 mismatch

**Comment by TNTran92** [comment]
> The correct url has now revealed a SHA256 mismatch

I'm still new to this. But how do you tell it's SHA mismatch?

**Comment by TNTran92** [comment]
> 



> It seems the link to the github release assets changed. The assets are now under https://github.com/alan-turing-institute/sktime/archive/refs/tags/v0.13.1.tar.gz url so the template in the `meta.yml` would need to be updated.

I can't find this directory in sktime's [Github](https://github.com/alan-turing-institute/sktime). How did you find this?

**Comment by lmmentel** [comment]
This is one of the release assets. Go to the release page for the latest release and it should have two assets attached, a zip file and a tar.gz file.

**Comment by lmmentel** [comment]
Looks like good progress :+1: 

It seems that there are still issues with dependency resolution:

```bash
Mamba failed to solve:
 - sktime-all-extras 0.13.1.* ha770c72_0
 - dtw-python
 - esig 0.9.7.*
 - filterpy >=1.4.5
 - gluonts >=0.9.0
 - kotsu >=0.3.1
 - matplotlib-base >=3.3.2
 - mne
 - numba >=0.53
 - pmdarima >=1.8.0,!=1.8.1
 - pykalman >=0.9.5
 - pyod >=0.8.0
 - python
 - scikit-posthocs >=0.6.5
 - seaborn >=0.11.0
 - sktime >=0.13.1,<0.13.2.0a0
 - statsforecast >=0.5.2
 - stumpy >=1.5.1
 - tbats >=1.1.0
 - tensorflow
 - tensorflow-probability
 - tsfresh >=0.17.0
 - tslearn >=0.5.2
 - python

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides esig 0.9.7 needed by sktime-all-extras-0.13.1-ha770c72_0
-   - nothing provides requested esig 0.9.7.*
-   - nothing provides requested gluonts >=0.9.0
-   - nothing provides requested kotsu >=0.3.1
```
Mamba does not provide the most helpful message but maybe you could try making it work locally first?

**Comment by TNTran92** [comment]
when built locally in Linux, the following exception was caught:

`failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: {'esig=0.9.7', "gluonts[version='>=0.9.0']", "kotsu[version='>=0.3.1']"}
WARNING:conda_build.build:failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: {'esig=0.9.7', "gluonts[version='>=0.9.0']", "kotsu[version='>=0.3.1']"}
Tests failed for sktime-all-extras-0.13.1-0.tar.bz2 - moving package to /home/ttran/anaconda3/conda-bld/broken
WARNING:conda_build.build:Tests failed for sktime-all-extras-0.13.1-0.tar.bz2 - moving package to /home/ttran/anaconda3/conda-bld/broken
TESTS FAILED: sktime-all-extras-0.13.1-0.tar.bz2
`

**Comment by TNTran92** [comment]
I think I've found the problem. On Pypi, the `meta.yaml` is looking into conda and/or conda-forge channel for `kotsu`, `gluonts` and 'esig.' However, these are not available on conda. At least in `gluonts` case, it's only available up to ver. 0.8.1 Fixing `gluonts` to 0.8.1 in the `meta.yaml` make that message go away for `gluonts`

Next, I'll try removing these 3 packages from the recipe. If it works, we may have to add a note to users that they'll have to manually install these using `pip`

**Comment by TNTran92** [comment]
Removing those 3 packages introduces a new failure message:

```
Traceback (most recent call last):
  File "/home/ttran/anaconda3/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/cli/main_build.py", line 488, in main
    execute(sys.argv[1:])
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/cli/main_build.py", line 477, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/build.py", line 3099, in build_tree
    test(pkg, config=metadata.config.copy(), stats=stats)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/build.py", line 2865, in test
    actions = environ.get_install_actions(metadata.config.test_prefix,
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda_build/environ.py", line 799, in get_install_actions
    actions = install_actions(prefix, index, specs, force=True)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/common/io.py", line 86, in decorated
    return f(*args, **kwds)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/plan.py", line 473, in install_actions
    txn = solver.solve_for_transaction(prune=prune, ignore_pinned=not pinned)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/core/solve.py", line 152, in solve_for_transaction
    unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/core/solve.py", line 195, in solve_for_diff
    final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/core/solve.py", line 313, in solve_final_state
    ssc = self._add_specs(ssc)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/core/solve.py", line 707, in _add_specs
    conflicts = ssc.r.get_conflicting_specs(tuple(MatchSpec(_)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/resolve.py", line 1106, in get_conflicting_specs
    C = r2.gen_clauses()
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/common/io.py", line 86, in decorated
    return f(*args, **kwds)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/resolve.py", line 909, in gen_clauses
    for ms in self.ms_depends(prec):
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/resolve.py", line 793, in ms_depends
    deps = [MatchSpec(d) for d in prec.combined_depends]
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/models/records.py", line 340, in combined_depends
    result = {ms.name: ms for ms in MatchSpec.merge(self.depends)}
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/models/match_spec.py", line 476, in merge
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/models/match_spec.py", line 476, in <lambda>
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/models/match_spec.py", line 507, in _merge
    final = this_component.merge(that_component)
  File "/home/ttran/anaconda3/lib/python3.9/site-packages/conda/models/match_spec.py", line 758, in merge
    raise ValueError("Incompatible component merge:\n  - %r\n  - %r"
ValueError: Incompatible component merge:
  - 'mpi_mpich_*'
  - 'mpi_mpich_tempest*'
```

**Comment by TNTran92** [comment]
Found a solution [here](https://github.com/conda/conda/issues/11442)

What it does is we need to define an environment.yaml that declares all dependencies and their channel (conda or pip)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -61,23 +61,6 @@ outputs:
         - sktime.transformations
         - sktime.utils
 
-  - name: sktime-all-extras
```
Comment (lmmentel) [inline recipe/meta.yaml:64]: This output needs to stay as it provides the `sktime-all-extras` conda package with full set of soft dependencies. Please revert the change.

**Review by lmmentel — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (TNTran92) [reply recipe/meta.yaml:64]: Added it back!

**Comment by dhirschfeld** [comment]
Closing as superseded.
